src/binding.cc

```diff
@@ -0,0 +1,162 @@
+      "wstring and u16string have the same value type on Windows");
+  std::u16string u16 = value.ToString();
+  return std::wstring(u16.begin(), u16.end());
+}
+
+Value ExportAllCertificatesSync(const CallbackInfo& args) {
+  std::wstring sys_store_name = MultiByteToWideChar(args[0].ToString());
+  DWORD store_type = args[1].ToNumber().Uint32Value();
+
+  try {
+    return BufferListToArray(
+        args.Env(), ExportAllCertificates(sys_store_name, store_type));
+  } catch (const std::exception& e) {
+    throw Error::New(args.Env(), e.what());
+  }
+}
+
+ExportCertificateAndKeyArgs GatherExportCertificateAndKeyArgs(const CallbackInfo& args) {
+  ExportCertificateAndKeyArgs exp_args;
+  exp_args.password_buf = MultiByteToWideChar(args[0].ToString());
+  exp_args.sys_store_name = MultiByteToWideChar(args[1].ToString());
+  exp_args.store_type = args[2].ToNumber().Uint32Value();
+  exp_args.require_private_key = args[4].ToBoolean();
+
+  Object search_spec = args[3].ToObject();
+  if (search_spec.HasOwnProperty("thumbprint")) {
+    exp_args.use_thumbprint = true;
+    Buffer<BYTE> thumbprint = search_spec.Get("thumbprint").As<Buffer<BYTE>>();
+    exp_args.thumbprint = {thumbprint.Data(), thumbprint.Data() + thumbprint.Length()};
+  } else if (search_spec.HasOwnProperty("subject")) {
+    exp_args.use_thumbprint = false;
+    exp_args.subject = MultiByteToWideChar(search_spec.Get("subject").ToString());
+  } else {
+    throw Error::New(args.Env(), "Need to specify either `thumbprint` or `subject`");
+  }
+  return exp_args;
+}
+
+// Export a given certificate from a system certificate store,
+// identified either by its thumbprint or its subject line.
+Value ExportCertificateAndKeySync(const CallbackInfo& args) {
+  ExportCertificateAndKeyArgs exp_args = GatherExportCertificateAndKeyArgs(args);
+  try {
+    auto result = ExportCertificateAndKey(exp_args);
+    return Buffer<BYTE>::Copy(args.Env(), result.data(), result.size());
+  } catch (const std::exception& e) {
+    throw Error::New(args.Env(), e.what());
+  }
+}
+
+Value ExportAllCertificatesAsync(const CallbackInfo& args) {  
+  class Worker final : public AsyncWorker {
+    public:
+      Worker(Function callback, std::wstring&& sys_store_name, DWORD store_type)
+        : AsyncWorker(callback, PACKAGE ":ExportAllCertificates"),
+          sys_store_name(std::move(sys_store_name)), store_type(store_type) {}
+      ~Worker() {}
+
+      void Execute() override {
+        results = ExportAllCertificates(sys_store_name, store_type);
+      }
+
+      void OnOK() override {
+        try {
+          Callback().Call({Env().Null(), BufferListToArray(Env(), results)});
+        } catch (const std::exception& e) {
+          throw Error::New(Env(), e.what());
+        }
+      }
+
+    private:
+      std::vector<std::vector<BYTE>> results;
+      std::wstring sys_store_name;
+      DWORD store_type;
+  };
+
+  Worker* worker = new Worker(
+      args[2].As<Function>(),
```

Comment:
This all works like JS does – `args[n]` is the JS `undefined` value if `n` is out of bounds, and then subsequently calling it will fail with an `undefined is not a function`-style JS exception.

So yeah, typechecking here would make sense if it weren't internal code that's wrapped externally anyway.